R/utils-paths.R

```diff
@@ -30,6 +32,30 @@ unsplit_path <- function(...) {
   gsub("^/*", "", file.path(...))
 }
 
+## partitions path into
+##   * name = substring after the last `/`
+##   * parent = substring up to the last `/`, processed with rootize_path()
+## if there is no `/`, put the input into `name`
+partition_path <- function(path) {
```

Comment:
There is a noticeable amount of repetition in several functions now, that `split_path_name()` did address. I still feel like smaller is better. But we might be able to build a helper that does more than `partition_path()`. Some sort of compromise.